mutect2_processes.nf

```diff
@@ -4,7 +4,6 @@
  */
 process mutect2 {
     tag "${patient}_${T}_${N}_${chromosome}_${start}_${end}"
-    cpus 4 // seems to use about 400% CPU per job
```

Comment:
Might be helpful to put in an iterative safety net here with increasing RAM.   We used to see some issues in regions with super-high coverage needing extra RAM.   Not sure if it will ever need more than 48Gb now that we don't process the decoy sequence, but might be something to keep an eye on.

Reply:
The iterative memory increase is a good idea, I'm just hesitant to include it *here* for a couple reasons:

Since this tool is public repo and (in theory) anyone could run it on their own hardware with their own inputs, we don't want to make too many assumptions about how much cpu/memory is needed or available for their runs. 

Once these changes are merged it should be trivial for anyone to change the cpu/memory by just tweaking their copy of the `nextflow.config` file. This is what we'll be doing at the GSC - we have our own "production" config so we'll just add the iterative retry to that.

The other reason to avoid iterative retry is that you can end missing errors very easily. If there's a bug in the code and it fails, the workflow will retry with more memory even though memory wasn't the issue, and you just end up wasting a bunch of extra time and resources.

Reply:
Thanks @mattdoug604.  Makes total sense.

Perhaps if we do run into memory requirements based on sample in the future we'll find the iterative increase helpful. (ie. some samples have more bias than results in super high coverage in some regions - ie. FFPE).  However, based on your comments above I agree that just leaving the control to the user.

Maybe in future releases we can add a config variable that controls any iterative increases in resource request.   I wouldn't suggest acting on this until we see evidence of it being of use.

Reply:
That sounds good, @RichardCorbett. I think adding it the iterative increase when we have a clear requirement/use case for it would be fine.